### PR TITLE
Integrating ChannelId, ChannelGroup

### PR DESCRIPTION
**Pull request by Aaronontheweb**
Integrating #40 with #11

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
Will rebase at the end

File: src/DotNetty.Transport/Channels/DefaultChannelId.cs [hunk 1]
```diff
@@ -0,0 +1,416 @@
+﻿using DotNetty.Buffers;
```
Comment (nayato) [inline src/DotNetty.Transport/Channels/DefaultChannelId.cs]: missing header, pls move usings inside namespace

File: src/DotNetty.Common/Utilities/SystemPropertyUtil.cs [hunk 2]
```diff
@@ -0,0 +1,105 @@
+﻿using System;
```
Comment (nayato) [inline src/DotNetty.Common/Utilities/SystemPropertyUtil.cs]: missing header, move usings inside namespace

File: src/DotNetty.Common/Utilities/StringUtil.cs [hunk 3]
```diff
@@ -0,0 +1,322 @@
+﻿using System;
```
Comment (nayato) [inline src/DotNetty.Common/Utilities/StringUtil.cs]: missing header, move usings inside namespace

File: src/DotNetty.Transport/Channels/Groups/ChannelGroupException.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
+﻿using System;
```
Comment (nayato) [inline src/DotNetty.Transport/Channels/Groups/ChannelGroupException.cs]: missing header, move usings inside namespace

Comment (nayato) [inline src/DotNetty.Transport/Channels/Groups/ChannelGroupException.cs]: missing header, move usings inside namespace

File: src/DotNetty.Transport/Channels/Groups/ChannelMatchers.cs [hunk 5]
```diff
@@ -0,0 +1,139 @@
+﻿using System;
```
Comment (nayato) [inline src/DotNetty.Transport/Channels/Groups/ChannelMatchers.cs]: missing header, move usings inside namespace

File: src/DotNetty.Transport/Channels/Groups/CombinedEnumerator.cs [hunk 6]
```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections.Generic;
```
Comment (nayato) [inline src/DotNetty.Transport/Channels/Groups/CombinedEnumerator.cs]: missing header, move usings inside namespace

File: src/DotNetty.Transport/Channels/Groups/DefaultChannelGroup.cs [hunk 7]
```diff
@@ -0,0 +1,387 @@
+﻿using DotNetty.Buffers;
```
Comment (nayato) [inline src/DotNetty.Transport/Channels/Groups/DefaultChannelGroup.cs]: missing header, move usings inside namespace

File: src/DotNetty.Transport/Channels/Groups/DefaultChannelGroupCompletionSource.cs [hunk 8]
```diff
@@ -0,0 +1,146 @@
+﻿using DotNetty.Common.Concurrency;
```
Comment (nayato) [inline src/DotNetty.Transport/Channels/Groups/DefaultChannelGroupCompletionSource.cs]: missing header, move usings inside namespace

File: src/DotNetty.Transport/Channels/Groups/IChannelGroup.cs [hunk 9]
```diff
@@ -0,0 +1,45 @@
+﻿using System;
```
Comment (nayato) [inline src/DotNetty.Transport/Channels/Groups/IChannelGroup.cs]: missing header, move usings inside namespace

File: src/DotNetty.Transport/Channels/Groups/IChannelGroupTaskCompletionSource.cs [hunk 10]
```diff
@@ -0,0 +1,20 @@
+﻿using System.Collections.Generic;
```
Comment (nayato) [inline src/DotNetty.Transport/Channels/Groups/IChannelGroupTaskCompletionSource.cs]: missing header, move usings inside namespace

File: src/DotNetty.Transport/Channels/IChannelId.cs [hunk 11]
```diff
@@ -0,0 +1,12 @@
+﻿
```
Comment (nayato) [inline src/DotNetty.Transport/Channels/IChannelId.cs]: missing header, move usings inside namespace

**Comment by Aaronontheweb** [comment]
@nayato added a script for adding missing headers to the `tools` folder, based on the Akka.NET one. Worked great!

**Comment by Aaronontheweb** [comment]
Once this pases review I'm going to rebase into two commits

1. One for @bobanco's original changes
2. For the integration / merging stuff I just did

Does that work for you @nayato?

**Comment by nayato** [comment]
@Aaronontheweb yep, looks good.

**Comment by Aaronontheweb** [comment]
@nayato all finished

**Pull request merged**